aptos-move/aptos-vm/src/aptos_vm_impl.rs

```diff
@@ -511,32 +508,25 @@ impl AptosVMImpl {
         module: &ModuleId,
         abort_code: u64,
     ) -> Option<AbortInfo> {
-        let entry = self
-            .metadata_cache
-            .entry(module.clone())
-            .or_insert_with(|| {
-                if let Some(m) = self
-                    .move_vm
-                    .get_module_metadata(module.clone(), &APTOS_METADATA_KEY)
-                {
-                    bcs::from_bytes::<RuntimeModuleMetadata>(&m.value).ok()
-                } else {
-                    None
-                }
-            });
-        if let Some(m) = entry.value() {
+        if let Some(m) = self.extract_module_metadata(module) {
             m.extract_abort_info(abort_code)
         } else {
             None
         }
     }
 
+    pub(crate) fn extract_module_metadata(
+        &self,
+        module: &ModuleId,
+    ) -> Option<RuntimeModuleMetadataV1> {
+        aptos_framework::get_vm_metadata(&self.move_vm, module.clone())
+    }
+
     pub fn new_session<'r, R: MoveResolverExt>(
         &self,
         r: &'r R,
         session_id: SessionId,
     ) -> SessionExt<'r, '_, R> {
-        self.metadata_cache.clear();
```

Comment:
Why do we need to change the caching behavior here?

Reply:
Just came along this, this was basically deactivated a while ago in a hotfix (flushed every time), and I decided to now remove it instead of investing the effort to bring it back, as @gelash suggested this time back.